packages/package-manager/src/utils/shouldUseYarn.ts

```diff
@@ -1,11 +1,19 @@
 import { execSync } from 'child_process';
 
+import { isUsingNpm } from './nodeWorkspaces';
+
 export default function shouldUseYarn(): boolean {
   try {
     if (process.env.npm_config_user_agent?.startsWith('yarn')) {
       return true;
     }
+
     execSync('yarnpkg --version', { stdio: 'ignore' });
+
+    if (isUsingNpm(process.cwd())) {
```

Comment:
Why verify that yarn is installed just to throw the result away and check for the lock file?  Maybe move this before the execSync call. Also since you're here, move that and the `if (process...` outside of the try/catch block since the try is for the execSync but it might suppress useful warnings from the other two statements.